### PR TITLE
Fix return type of `has_sorted_indices` of sparse array

### PR DESCRIPTION
**Pull request by asi1024**
```py
>>> import scipy.sparse
>>> scipy.sparse.random(3, 4).tocsc().has_sorted_indices
1
>>> import cupyx.scipy.sparse
>>> cupyx.scipy.sparse.random(3, 4).tocsc().has_sorted_indices
True
```

### CODE REVIEW

**Comment by toslunar** [comment]
I posted scipy/scipy#13408.

**Comment by asi1024** [comment]
@toslunar Should we add a workaround in the test code instead?

**Comment by toslunar** [comment]
scipy/scipy#13408 is fixed :tada:

> Should we add a workaround in the test code instead?

Yes.  Even though the fixed SciPy hasn't been released, I'd edit the test rather than changing the behavior back and forth.

**Comment by asi1024** [comment]
@toslunar PTAL!

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_csc.py [hunk 1]
```diff
@@ -890,11 +890,20 @@ def test_has_canonical_format4(self, xp, sp):
         assert 2 == len(M.indices)  # unaffected content
         return M
 
+    @testing.with_requires('scipy>1.6.0')
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_csc.py:893]: Maybe `scipy>=1.7`?

Comment (asi1024) [reply tests/cupyx_tests/scipy_tests/sparse_tests/test_csc.py:893]: Won't this SciPy's bug be fixed in 1.6.1?

Comment (toslunar) [reply tests/cupyx_tests/scipy_tests/sparse_tests/test_csc.py:893]: You are right.  The SciPy PR has the `backport-candidate` label.

**Comment by toslunar** [comment]
pfnCI, test this please.

**Review by toslunar — APPROVED** [review]

**Pull request merged**